src/patterns/security-code/index.md.njk

```diff
@@ -0,0 +1,111 @@
+---
+title: Security code
+description: Identifying users when they sign in
+section: Patterns
+theme: Ask users for…
+aliases: 2FA
+backlog_issue_id: 25
+layout: layout-pane.njk
+---
+
+{% from "_example.njk" import example %}
+
+Use this pattern to verify users via a security code sent to their phone.
+
+{{ example({group: "patterns", item: "security-code", example: "default", html: true, nunjucks: true, open: false}) }}
+
+## When to use this pattern
+
+Use this pattern if your users need to sign in to your service.
+
+Requiring users to enter a security code sent to their mobile phone provides a second layer of security over a [password](/patterns/passwords/).
+
+You can require a security code on every sign in or only once per device, depending upon the risk of your service.
+
+## How it works
+
+You should first ask users for their [password](/patterns/passwords/). Once they have successfully entered this, send a text message to their mobile phone with a 5 digit code in this format:
+
+{% from "govuk/components/inset-text/macro.njk" import govukInsetText %}
+{{ govukInsetText({
+  text: "12345 is your [service name] security code"
```

Comment:
There’s actually a proposal backed by Apple and Google to standardise the format of text messages containing security codes so that it’s easy for browsers to automatically extract them and fill them in, while still being human readable:

- https://github.com/wicg/sms-one-time-codes
- https://wicg.github.io/sms-one-time-codes/

Reply:
Interesting. They both suggest this format, appending the domain of the service on a new line prefixed with an `@`:

```
12345 is your [service name] security code

@example.com #747723
```

My only concern would be whether this confuses users or not. Perhaps this is one where we should "wait and see" how this gets adopted. I have found it frustrating though where the Mac / iPhone is unable to auto-suggest the code for some services – that's the downside with using heuristics.

Reply:
We should definitely nab [`autocomplete="one-time-code"`](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#attr-fe-autocomplete-one-time-code) (linked to from that proposal).

Reply:
It’s not entirely clear from the proposal but the it’s only the last line that matters. The rest can be anything we like. I agree it’s probably too early to implement it yet.

Looks like `autocomplete="one-time-code"` is already in the PR.

Reply:
`autocomplete="one-time-code"` is already in this PR, and we've been using that in production. I've no idea if any browsers are actually making use of it yet though...

Reply:
Apparently Apple is now supporting the standardised format (which they call “domain-bound codes”): https://developer.apple.com/news/?id=z0i801mg

Reply:
@alexbishop1 I've added a section on domain-bound codes here: https://deploy-preview-1265--govuk-design-system-preview.netlify.app/patterns/security-code/#how-it-works

Possibly it shouldn't be included _yet_ though, as it's still quite a new and unfamiliar standard?